app/controllers/posts_controller.rb

```diff
@@ -7,5 +7,30 @@ def index
   def show
     @post = Post.includes(:comments).find(params[:id])
     @user = User.find(params[:user_id])
+    @likes = @post.likes.all
+  end
+
+  def new
+    @user = current_user
```

Comment:
- [ ] @cosywasswa Kindly consider removing this code, this is because you are only using it in the create method and this may lead to having a redundant code that doesn't have any use in our codebase.